### PR TITLE
Fix many minor bugs in several scripts

### PR DESCRIPTION
**Pull request by hugsy**
## Description/Motivation/Screenshots

restored to a workable state - for 2024.01 :
 - `remote`
 - `windbg`
 - `assemble`
 - `libc_function_args` is completely broken
 - 

## How Has This Been Tested ?

"Tested" indicates that the PR works *and* the unit test (i.e. `make test`) run passes without issue.

*  [x] x86-32
*  [x] x86-64
*  [ ] ARM
*  [ ] AARCH64
*  [ ] MIPS
*  [ ] POWERPC
*  [ ] SPARC
*  [ ] RISC-V

## Checklist

*  [x] My code follows the code style of this project.
*  [x] My change includes a change to the documentation, if required.
*  [x] If my change adds new code,
   [adequate tests](https://hugsy.github.io/gef/testing) have been added.
*  [x] I have read and agree to the
   [CONTRIBUTING](https://github.com/hugsy/gef/blob/main/.github/CONTRIBUTING.md) document.

### CODE REVIEW

File: scripts/capstone.py [hunk 1]
```diff
@@ -103,20 +137,32 @@ class CapstoneDisassembleCommand(GenericCommand):
...
         ctx = gef.gdb.commands["context"]
         assert isinstance(ctx, ContextCommand)
-        if gef.config[f"{self._cmdline_}.use-capstone"] == True:
+        if gef.config[f"{self._cmdline_}.use-capstone"]:
             self.__original_disassembler = ctx.instruction_iterator
             ctx.instruction_iterator = cs_disassemble
         else:
             # `use-capstone` set to False
-            if ctx.instruction_iterator == cs_disassemble and self.__original_disassembler:
+            if (
```
Comment (Grazfather) [inline scripts/capstone.py:162]: Do you prefer this style (four lines) over either 1 line or 2 with parens or 2 with `\`?

File: scripts/capstone.py [hunk 2]
```diff
@@ -141,7 +189,9 @@ def do_invoke(self, _: List[str], **kwargs: Any) -> None:
 
         insns = []
         opcodes_len = 0
-        for insn in cs_disassemble(location, length, skip=length * self.repeat_count, **kwargs):
+        for insn in cs_disassemble(
+            location, length, skip=length * self.repeat_count, **kwargs
+        ):
```
Comment (Grazfather) [inline scripts/capstone.py:194]: I don't think we need (nor do I like) the hanging parens. Wastes lines imo.
```suggestion
        for insn in cs_disassemble(
            location, length, skip=length * self.repeat_count, **kwargs):
```

**Review by Grazfather — APPROVED** [review]
Style Q, fine otherwise.

File: scripts/capstone.py (hunk 1 above)
Comment (hugsy) [reply scripts/capstone.py:162]: I almost forgot: I started using `black` for gef-extras (only for now), and those kind of output were the result. I'm not sure how I feel yet about it.

Comment (Grazfather) [reply scripts/capstone.py:162]: Ah yeah, that may be why I didn't like black (remember I said it was too opinonated and I didn't like the opinion).

I am 100% for auto-formatting everything! That will be nice and easy to check. I don't love black but I could live with it.

**Pull request merged**